mcs/class/System/System.Net/HttpWebRequest.cs

```diff
@@ -1184,6 +1167,20 @@ public override IAsyncResult BeginGetResponse (AsyncCallback callback, object st
 			if (Aborted)
 				throw CreateRequestAbortedException ();
 
+			string transferEncoding = TransferEncoding;
+			if (!sendChunked && transferEncoding != null && transferEncoding.Trim () != "") {
+				/*
```

Comment:
This section here https://github.com/baulig/mono/blob/02c139772082bd708b510c0732bda27bc7bc936b/mcs/class/System/System.Net/HttpWebRequest.cs#L660 already matches the CoreFX behavior.

We can also get rid of the `Trim()` calls once we have switched `HttpClient` to the CoreFX version and also use the header code from CoreFX.